src/init.cpp

```diff
@@ -1599,10 +1599,18 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
             vImportFiles.push_back(strFile);
     }
     threadGroup.create_thread(boost::bind(&ThreadImport, vImportFiles));
-    if (chainActive.Tip() == NULL) {
-        LogPrintf("Waiting for genesis block to be imported...\n");
-        while (!fRequestShutdown && chainActive.Tip() == NULL)
+
+    // Wait for genesis block to be processed
+    bool fHaveGenesis = false;
+    while (!fHaveGenesis && !fRequestShutdown) {
+        {
+            LOCK(cs_main);
+            fHaveGenesis = (chainActive.Tip() != NULL);
```

Comment:
nit:   There exists a CChain method `Genesis` that returns `NULL` in the case that the `size()` of its `vChain` member is `0`.

This seems like the right case to call that method, since we're just checking that the return is non `NULL`.

Reply:
Possibly, but if we change that here then we should also change it in `ThreadNotifyWallets`. Since this PR is intended to be the minimum sufficient bugfix, and the current code on this PR is correct, I'm reluctant to change it.